server/routes/api/team.ts

```diff
@@ -60,4 +62,65 @@ router.post("team.update", auth(), async (ctx) => {
   };
 });
 
+router.post("team.create", auth(), async (ctx) => {
+  const { user } = ctx.state;
+  const { name } = ctx.body;
+  assertLength(name, 2, "Name must be 2 or more characters");
+
+  const existingTeam = await Team.scope("withAuthenticationProviders").findByPk(
+    user.teamId
+  );
+
+  authorize(user, "createTeam", existingTeam);
+
+  const authenticationProviders = existingTeam?.authenticationProviders.map(
+    (provider) => {
+      return {
+        name: provider.name,
+        providerId: provider.providerId,
+      };
+    }
+  );
+
+  invariant(
+    authenticationProviders?.length,
+    "authentication provideers must exist"
+  );
+
+  const team = await teamCreator({
+    name,
+    subdomain: name,
+    authenticationProviders,
+    ip: ctx.ip,
+  });
+
+  const newUser = await User.create({
```

Comment:
Should really be a transaction wrapping around this and the event creation so that one can't be committed without the other

Reply:
Still missing transaction